omero/developers/PythonBlitzGateway.rst

```diff
@@ -80,6 +80,41 @@ loading).
        survivin
     >>> conn.close()
 
+
+Unicode
+"""""""
+
+When unwrapping ``omero.rtypes.rstring`` attributes from
+``omero.model.object`` instances as shown above,
+the ``BlitzObjectWrapper`` subclasses will decode the utf8-encoded
+string and return the resulting unicode string.
+
+However, some methods added to ``BlitzObjectWrappers``
+will return regular strings as follows::
+
+    BlitzObjectWrapper.getName()            except ExperimenterWrapper.getName() and LogicalChannelWrapper.getName()
```

Comment:
why such diff?

Reply:
LogicalChannel.getName() returns unicode: https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroPy/src/omero/gateway/__init__.py#L7294
Same for Experimenter: https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroPy/src/omero/gateway/__init__.py#L5777

Here I am just documenting the current behaviour, not trying to explain why it's this way.

Reply:
I understand but why such a diff in the first place (if you remember)

Reply:
I don't think there's any intention to have some methods return string and some return unicode, but this is how it's been since the BlitzGateway was created:
```
commit f0311e5fd641f7bb5c7ff47995fdb2dc8e06163d (HEAD)
Author: carlos <carlos@05709c45-44f0-0310-885b-81a1db45b4a6>
Date:   Mon Apr 6 14:08:44 2009 +0000
    Adding omero.gateway, a python wrapper for omero.client.
```
This returns String
```
def getName (self):
        return self._obj.getName().val
```
This returns unicode
```
    def __getattr__ (self, attr):
        if hasattr(self._obj, attr):
            rv = getattr(self._obj, attr)
            if hasattr(rv, 'val'):
                return isinstance(rv.val, StringType) and rv.val.decode('utf8') or rv.val
```
And in all the time I've used the BlitzGateway, I've never really been aware that some return unicode.

Reply:
Maybe we should record and start an audit  so we can deprecate method and unify that

Reply:
I'm not sure we can deprecate ```getName()```, since we can't remove it, we can only change the return type to Unicode. We could maybe just add a warning that the return type will change to Unicode at some point (5.5.0)? We should probably also make every BlitzGateway method return Unicode, or all return String. String is easier since that's what we're starting with everywhere.

Reply:
If we're going to do that review and unification _anyway_, then I'd go back to the suggestion of making it user-configurable.

Reply:
I take it this discussion does not affect me merging this PR @joshmoore ?

Reply:
Correct.